### PR TITLE
Turin RAS enablement

### PR DESCRIPTION
**Pull request by abinayaddhandapani**
1. Enable runtime error polling for MCA, DRAM CECC and PCIE errors.
2. Reorganise the code and move the functionalities to specific cpp files
3. Add new DEBUG_LOG_ID's for the Turin platform
4. Trigger cold reset via RSMRST toggling
5. Support to download CPER files via redfish for runtime and fatal errors

### CODE REVIEW

File: inc/cper.hpp [hunk 1]
```diff
@@ -154,9 +187,9 @@ struct error_section_descriptor {
   uint8_t                            Reserved;
   uint32_t                           SectionFlags;
   GUID_T                             SectionType;
-  GUID_T                             FRUId;
+  uint64_t                           FRUId[INDEX_2];
   uint32_t                           Severity;
-  unsigned char                      FRUText[20];
+  char                               FRUText[20];
```
[the remark this answers is not in the document]
Comment (supven01) [reply inc/cper.hpp:192]: nit: tab used instead of spaces or vice versa

**Review by supven01 — CHANGES_REQUESTED** [review]
The re-organization looks much better now especially the configuration.hpp and configuration.cpp.
There are still enormous files like main.cpp and fatal_error.cpp which needs to be refactored after 06 Turin release.

File: inc/write_cper_data.hpp [hunk 2]
```diff
@@ -0,0 +1,268 @@
+#include "ras.hpp"
```
[the remark this answers is not in the document]
Comment (abinayaddhandapani) [reply inc/write_cper_data.hpp:1]: defenitions in write_cper_data.hpp is moved to write_cper_data.cpp file. write_cper_data.hpp is no longer needed and is check-in by mistake. Will delete this file and push the code again

File: inc/Config.hpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
+#include "ras.hpp"
+
+class Configuration
+{
```
[the remark this answers is not in the document]
Comment (abinayaddhandapani) [reply inc/Config.hpp:4]: Ack

File: service_files/com.amd.crashdump.service [hunk 4]
```diff
@@ -7,6 +7,9 @@ ExecStart=/usr/bin/amd-ras
 SyslogIdentifier=amd-ras
 Type=dbus
 BusName=com.amd.crashdump
+Restart=always
+TimeoutStartSec=infinity
+ExecStartPre=/bin/sleep 5
```
[the remark this answers is not in the document]
Comment (abinayaddhandapani) [reply service_files/com.amd.crashdump.service:12]: When the application is restarted due to any crash, without the sleep , the service is throwing error as "Service start request repeated too quickly, refusing to start limit"

File: inc/cper.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (abinayaddhandapani) [reply inc/cper.hpp]: Ack

File: inc/cper.hpp [hunk 5]
```diff
@@ -194,8 +218,7 @@ struct context_info {
   CRASHDUMP_T                        CrashDumpData[GENOA_MCA_BANKS];
   DF_DUMP                            DfDumpData;
   uint32_t                           Reserved[96];
```
[the remark this answers is not in the document]
Comment (abinayaddhandapani) [reply inc/cper.hpp]: Ack

File: inc/cper_runtime.hpp [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+struct pcie_error_section {
+    uint64_t               ValidationBits;
+    uint32_t               PortType;
+    pcie_version           Version;
+    uint32_t               CommandStatus;
+    uint32_t               Reserved;
+    device_id              DeviceId;
+    uint64_t               DeviceSerialNumber;
+    uint32_t               BridgeControlStatus;
+    uint8_t                Capability[60];
```
[the remark this answers is not in the document]
Comment (abinayaddhandapani) [reply inc/cper_runtime.hpp]: Ack

File: CMakeLists.txt [hunk 7]
```diff
@@ -20,7 +20,13 @@ set(DBUS_INTF_NAME "com.amd.crashdump")
 add_definitions(-DDBUS_OBJECT_NAME="/${DBUS_OBJECT_NAME}")
 add_definitions(-DDBUS_ENTRY_NAME="/${DBUS_ENTRY_NAME}")
 add_definitions(-DDBUS_INTF_NAME="${DBUS_INTF_NAME}")
-set(SRC_FILES src/main.cpp )
+set(SRC_FILES src/main.cpp 
+              src/runtime_errors.cpp 
+              src/write_cper_data.cpp 
+              src/dbus.cpp
```
[the remark this answers is not in the document]
Comment (abinayaddhandapani) [reply CMakeLists.txt]: Ack

**Review by supven01 — APPROVED** [review]

**Pull request merged**